ruby/bindings/InitRubyBindings.cpp

```diff
@@ -662,6 +662,10 @@ end)ruby";
     // in hash.c there's lot of env-related functions, but all private, aside from rb_env_clear();
     const std::string clearEnvs = R"ruby(
 ENV.reject! { |k, _| ['GEM', 'BUNDLE'].any? { |x| k.start_with?(x) } }
+# No env variable isn't the same as empty. Gem::PathSupport does stuff like @home = env["GEM_HOME"] || Gem.default_dir
+ENV['GEM_HOME'] = ''
+ENV['GEM_PATH'] = ''
+ENV['GEM_SPEC_CACHE'] = ''
```

Comment:
#5190 Do not leave GEM_HOME etc to be undefined, it's not the same as having a blank. + Call Gem::clear_paths

https://github.com/ruby/ruby/tree/v3_2_2lib/rubygems/path_support.rb#L8